server/src/pages/shot/editor.js

```diff
@@ -270,14 +270,22 @@ exports.Editor = class Editor extends React.Component {
         let moveX = movement[0];
         moveX = moveX == "*" ? ["x1", "x2"] : [moveX];
         for (let moveDir of moveX) {
-          selectedPos[moveDir] = this.truncateX(resizeStartSelected[moveDir] + diffX);
+          if (resizeDirection == "move" && (selectedPos.left == 0 && resizeStartSelected.left + diffX < 0 || selectedPos.right == this.canvasWidth && resizeStartSelected.right + diffX > this.canvasWidth)) {
+            break;
+          } else {
+            selectedPos[moveDir] = this.truncateX(resizeStartSelected[moveDir] + diffX);
+          }
         }
       }
       if (movement[1]) {
```

Comment:
I just noticed this `if` is indented one more level than the above `if`, which is a little confusing at a glace.